File: 3 Loops/7Nested_loop7_ButterFly.js
function printButterfly(n) {
    // Upper half of the butterfly
    for (let i = 1; i <= n; i++) {
        // Left part (i stars)
        let row = '*'.repeat(i);

        // Middle spaces (2 * (n - i) spaces)
        row += ' '.repeat(2 * (n - i));

        // Right part (i stars)
        row += '*'.repeat(i);

        console.log(row);
    }
    // Lower half of the butterfly
    for (let i = 1; i < n; i++) {
        // Left part (n - i stars)
        let row = '*'.repeat(n - i);

        // Middle spaces (2 * i - 1 spaces)
        row += ' '.repeat(2 * i - 1);

        // Right part (n - i stars)
        row += '*'.repeat(n - i);

        console.log(row);
    }
}
// Input: Number of rows in half butterfly
const n = 4;
printButterfly(n);

File: 3 Loops/7Nested_loop7_ButterFly.py
def print_butterfly(n):
    # Upper half of the butterfly
    for i in range(1, n + 1):
        # Left part (i stars)
        print('*' * i, end='')
        # Middle spaces (2 * (n - i) spaces)
        print(' ' * (2 * (n - i)), end='')
        # Right part (i stars)
        print('*' * i)
    # Lower half of the butterfly
    for i in range(1, n):
        # Left part (n - i stars)
        print('*' * (n - i), end='')
        # Middle spaces (2 * i - 1 spaces)
        print(' ' * (2 * i - 1), end='')
        # Right part (n - i stars)
        print('*' * (n - i))
# Input: Number of rows in half butterfly
n = 4
print_butterfly(n)
